src/prefect_server/api/runs.py

```diff
@@ -344,9 +344,33 @@ async def delete_flow_run(flow_run_id: str) -> bool:
     return bool(result.affected_rows)  # type: ignore
 
 
+@register_api("runs.update_flow_run_agent_instance")
+async def update_flow_run_agent_instance(
```

Comment:
This was another thing we discussed but it (along with the no index on the agent_instance_id on the flow run table) was decided to keep on the flow run table because in the information we are choosing to display a user is only going to want to see the agent instance which kicked off the most recent run

Reply:
Yea, I view the "multiple agents running one flow" pattern as a future enhancement, and in the meantime we can definitely store the agent instance ID on the Submitted state directly - in fact we can add the agent ID to the "Submitted" log for visibility